ubuntu/main.cpp

```diff
@@ -143,11 +110,11 @@ static int gst_pipeline_init(gst_app_t *app)
 
     gst_init(NULL, NULL);
 
-    const char* vid_launch_str = "appsrc name=mysrc is-live=true block=false max-latency=100000 do-timestamp=true ! video/x-h264, width=800,height=480,framerate=30/1 ! decodebin2 name=mydecoder "
+    const char* vid_launch_str = "appsrc name=mysrc is-live=true block=false max-latency=1000000 do-timestamp=true stream-type=stream typefind=true ! video/x-h264, width=800,height=400,framerate=30/1 ! h264parse ! avdec_h264 "
```

Comment:
That was a typo. I think the video/x-h264, width=800,height=400,framerate=30/1 capsfilter can be left out altogether, because typefind=true is there. That would also mean that we can use higher resolutions, but I'd need to test that.